### PR TITLE
feat(CDN): Support emoji size

### PR DESCRIPTION
**Pull request by Jiralite**
**Please describe the changes this PR makes and why it should be merged:**
`size` may affect emojis:

- https://cdn.discordapp.com/emojis/489818863341535247.gif?size=256
- https://cdn.discordapp.com/emojis/1117578868140806164.webp?size=16

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

**Review by didinele — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Review by sdanialraza — APPROVED** [review]

File: packages/rest/src/lib/CDN.ts [hunk 1]
```diff
@@ -154,14 +157,39 @@ export class CDN {
...
-		return this.makeURL(`/emojis/${emojiId}`, { extension });
+	// eslint-disable-next-line @typescript-eslint/unified-signatures
+	public emoji(emojiId: string, extension?: ImageExtension): string;
+
+	public emoji(emojiId: string, options?: ImageExtension | Readonly<BaseImageURLOptions>): string {
+		let resolvedOptions;
+
+		if (typeof options === 'string') {
+			if (!deprecationEmittedForEmoji) {
+				deprecationWarning('Passing a string for CDN#emoji() is deprecated. Use an object instead.');
```
Comment (ckohen) [inline packages/rest/src/lib/CDN.ts:183]: This isn't right is it? It's specifically for the second param, which this doesn't indicate to me

**Review by ckohen — CHANGES_REQUESTED** [review]

File: packages/rest/src/lib/CDN.ts (hunk 1 above)
Comment (Jiralite) [reply packages/rest/src/lib/CDN.ts:183]: Would "Passing a string for the extension..." be better?

Comment (ckohen) [reply packages/rest/src/lib/CDN.ts:183]: I don't think so, cause you still pass a string to the extension, it'd be more along the lines of "Passing an extension as the second parameter" or something (I don't like what I said there even cause it kinda suggests you can't pass an extensions at all)

Comment (Jiralite) [reply packages/rest/src/lib/CDN.ts:183]: This?

```suggestion
				deprecationWarning(
					'Passing a string for the second parameter of CDN#emoji() is deprecated. Use a sole object for the first parameter instead.',
				);

```

Comment (ckohen) [reply packages/rest/src/lib/CDN.ts:183]: Am I misunderstanding something, the first parameter is still the emojiId regardless right?

Comment (Jiralite) [reply packages/rest/src/lib/CDN.ts:183]: You're right! I've edited the suggestion.

Comment (ckohen) [reply packages/rest/src/lib/CDN.ts:183]: Yeah that's probably fine.

**Review by ckohen — APPROVED** [review]

**Pull request merged**